### PR TITLE
Dev workflow creation and Slurm job metrics additon 

### PR DESCRIPTION
**Pull request by vipulnayyar**
This PR has the following changes:

* Created a Dev workflow, that enabled developers to run monitoring dashboard in dev mode. The dev mode allows easy editing of Grafana dashboards.
* Added more Slurm job metrics to Grafana dashboard.

### CODE REVIEW

File: grafana/dashboards/slurm-job-metrics.json [hunk 1]
```diff
@@ -957,13 +1464,13 @@
     ]
   },
   "time": {
-    "from": "now-1h",
-    "to": "now"
+    "from": "2024-06-19T09:20:35.686Z",
```
Comment (gsangwell) [inline grafana/dashboards/slurm-job-metrics.json]: We've hard coded the `to` and `from` of this dashboard to a specific datetime. I believe this should be `now - 1hr` and `now`.

**Review by gsangwell — CHANGES_REQUESTED** [review]
image

The headings here could also do with changing to `Wait Time` + `Runtime` respectively to match the others.

I also believe this requires some changes to the `slurm-job-exporter` for the Wait Time metrics? Please could you do a related PR for that + reference here.

File: grafana/dashboards/slurm-job-metrics.json (hunk 1 above)
Comment (vipulnayyar) [reply grafana/dashboards/slurm-job-metrics.json]: Done

Comment (vipulnayyar) [reply grafana/dashboards/slurm-job-metrics.json]: @gsangwell , corresponding changes done to Slurm job exporter are visible in @shubhamdang 's request here at https://github.com/openflighthpc/slurm-job-exporter/pull/1

For `Wait Time` heading change, only relevant modification is required in the Grafana dashboard, as how it is done for other metrics

Comment (vipulnayyar) [reply grafana/dashboards/slurm-job-metrics.json]: @gsangwell , can you please have a look again for the requested changes?

**Pull request closed without merging**